core/source_map.rs

```diff
@@ -72,20 +77,70 @@ impl<G: SourceMapGetter> SourceMapper<G> {
       ext_source_maps: Default::default(),
       getter,
       stashed_file_name: Default::default(),
+      maybe_module_map: None,
     }
   }
 
+  pub fn set_module_map(&mut self, module_map: Rc<ModuleMap>) {
+    assert!(self.maybe_module_map.replace(module_map).is_none());
+  }
+
   pub fn has_user_sources(&self) -> bool {
     self.getter.is_some()
   }
 
+  pub fn apply_source_map_from_module_map(
+    &mut self,
+    scope: &mut v8::HandleScope,
+    file_name: &str,
+    line_number: u32,
+    column_number: u32,
+  ) -> Option<SourceMapApplication> {
+    let module_map = self.maybe_module_map.as_ref()?;
+    let id = module_map.get_id(file_name, RequestedModuleType::None)?;
+
+    let module_handle = module_map.get_handle(id).unwrap();
+    let module = v8::Local::new(scope, module_handle);
+    let unbound_module_script = module.get_unbound_module_script(scope);
+    let maybe_source_mapping_url =
+      unbound_module_script.get_source_mapping_url(scope);
+
+    // TODO(bartlomieju): This should be the last fallback and it's only useful
+    // for eval - probably for `Deno.core.evalContext()`.
+    let maybe_source_url = unbound_module_script
```

Comment:
See https://developer.chrome.com/blog/sourcemaps#sourceurl_and_displayname_in_action_eval_and_anonymous_functions for `sourceURL`